packages/core/admin/admin/src/hooks/useDocument.ts

```diff
@@ -0,0 +1,38 @@
+import { Schema, Entity as StrapiEntity, Attribute } from '@strapi/types';
+
+import { createYupSchema } from '../content-manager/utils/validation';
+
+export interface Entity {
+  id: StrapiEntity.ID;
+  createdAt: string;
+  updatedAt: string;
+}
+
+interface ValidateOptions {
+  contentType: Schema.ContentType;
+  components: {
+    [key: Schema.Component['uid']]: Schema.Component;
+  };
+}
+
+/**
+ * @alpha - This hook is not recommended to use it because is likely to be completely change, use it at your own risk
+ */
+export function useDocument() {
+  /**
+   * @TODO: Ideally, we should get the contentType and components schemas from the redux store
+   * But at the moment the store is populated only inside the content-manager by useContentManagerInitData
+   * So, we need to receive the content type schema and the components to use the function
+   */
+  const validate = (
+    entry: Entity & { [key: string]: Attribute.Any },
+    { contentType, components }: ValidateOptions
+  ) => {
+    // @ts-expect-error - @TODO: createYupSchema types need to be revisited
+    const schema = createYupSchema(contentType, { components }, { isCreatingEntry: false });
+
+    return schema.validateSync(entry, { abortEarly: false });
```

Comment:
`validateSync` throws if it fails right? What benefits do you see having this behaviour over catching the error, formatting it and returning it correctly? It creates a better documented behaviour imo as opposed to people having to handle the conversion to usable error messages themselves... thoughts?

Reply:
I think about this and I agree I would prefer to return the formatted error, but in that case I would change the name function from `validate` to `getValidationErrors` because is more explicit of what the function is doing, wdyt?

Reply:
I still think it's fine to call it validate, because you're validating the document – you might return errors, but you might return a success.

think of it like an API route imo?

Reply:
Originally I was imagining something like Josh suggested but I'm also ok with what you have here and letting the consumer handle the errors. 

Another thought would be to expose a separate function on the hook like `formatValidationErrors` that could compose `validate`. I'm not sure if it makes sense or not but I'm thinking something like:

`validate` returns a tuple `[isValid, errors]`, then `formatValidationErrors` can handle this result. 

```ts
formatValidationErrors(validate(entry, contentType))
```

Reply:
I think we're overcomplicating it – imo everything you need to do in releases, people will need to do. IF we find a need for composing the functions, _then_ we should support it imo always build it the best way for your use-case which afaik is, validate document, return me errors if there are some 😄

Reply:
Works for me. I'm good either way or a combination of the too :)